Public/Src/Engine/Scheduler/Fingerprints/ObservedInputProcessor.cs

```diff
@@ -892,11 +891,11 @@ private static bool IsPathWithinAny(PathTable pathTable, AbsolutePath path, IEnu
             where TTarget : struct, IObservedInputProcessingTarget<TObservation>
         {
             using (var pooledPathSet = Pools.GetAbsolutePathSet())
-            using (var pooledStringIdSet = Pools.GetStringIdSet())
             {
-                HashSet<StringId> accessedFileNames = pooledStringIdSet.Instance;
                 HashSet<AbsolutePath> visitedPaths = pooledPathSet.Instance;
 
+                HashSet<StringId> accessedFileNames = new HashSet<StringId>(pathTable.StringTable.CaseInsensitiveEqualityComparer);

```

Comment:
>pathTable.StringTable.CaseInsensitiveEqualityComparer [](start = 76, length = 53)

Is it OK for non-Windows?

Reply:
for now it is, and if we ever get to port this to Linux, we'll have to deal with this in a whole bunch of places

Reply:
this is the same behavior as we currently have (see `SearchPathDirectoryMembershipFilter`)

---
In reply to: [347749353](https://github.com/microsoft/BuildXL/pull/1236#discussion_r347749353) [](ancestors = 347749353)